src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs

```diff
@@ -1479,11 +1479,9 @@ .locals init (T1 V_0)
   IL_0018:  box        ""T2""
   IL_001d:  call       ""void System.Console.WriteLine(object)""
   IL_0022:  ldarg.0
-  IL_0023:  stloc.0
-  IL_0024:  ldloc.0
-  IL_0025:  box        ""T1""
-  IL_002a:  pop
-  IL_002b:  ret
+  IL_0023:  box        ""T1""
+  IL_0028:  pop
+  IL_0029:  ret
```

Comment:
This changed because the left operand of the coalesce (`t1`) has conversion to `T2` which is not a reference type (previously the check was against `T1` which is a reference type). I'm not actually sure why that's the case since there is `where T1 : T2` I would expect both to be detected as reference types.